File: 2022/CSharp/Day19.cs
namespace AdventOfCode.CSharp.Year2022;

public class Day19_Not_Enough_Minerals
{
    private const int DAY = 19;

    private readonly ITestOutputHelper output;
    public Day19_Not_Enough_Minerals(ITestOutputHelper output) => this.output = output;



    private (List<int> input, int? expected) GetTestData(int part, string inputName)
    {
        var input = InputHelper.ReadLines(DAY, inputName)
            .Select(l => l.ToInt32())
            .ToList();

        var expected = InputHelper.ReadText(DAY, $"{inputName}-answer{part}")
            ?.ToInt32();

        return (input, expected);
    }



    [Theory]
    [InlineData(1, "example1")]
    [InlineData(1, "input")]
    public void Part1(int part, string inputName)
    {
        var (input, expected) = GetTestData(part, inputName);

        var value = DoWork(input);

        output.WriteLine($"Answer: {value}");

        Assert.Equal(expected, value);
    }


    [Theory]
    [InlineData(2, "example1")]
    [InlineData(2, "input")]
    public void Part2(int part, string inputName)
    {
        var (input, expected) = GetTestData(part, inputName);

        var value = DoWork(input);

        output.WriteLine($"Answer: {value}");

        Assert.Equal(expected, value);
    }



    private int DoWork(List<int> input)
    {
        return -1;
    }
}

File: 2021/FSharp/Day19.fs
namespace AdventOfCode.FSharp.Year2021

open FSharp.Common
open System
open Xunit



module ``Day 19: Beacon Scanner`` =
    let day = 19



    //-------------------------------------------------------------------------



    type private Puzzle () =

        //How many beacons are there?
        member this.RunPart1 (input:string list) =
            let result = 0
            result


        // 
        //member this.RunPart2 (input:string list) =
        //    let result = 0
        //    result



    //-------------------------------------------------------------------------



    let private GetPuzzleInput (part:int) (name:string) =
        let input =
            InputHelper.LoadLines (2021, day, name)
            |> Seq.toList

        let answer = 
            InputHelper.LoadAnswer (2021, day, $"%s{name}-answer%i{part}")
            |> InputHelper.AsInt
        
        input, answer
    
    
    [<Theory>]
    [<InlineData("example1")>]
    [<InlineData("input")>]
    let Part1 (name:string) =
        let input, expected = GetPuzzleInput 1 name
    
        let actual = (new Puzzle()).RunPart1 input
    
        match expected with
        | None -> Assert.Null actual
        | _ -> Assert.Equal (expected.Value, actual)
    
    
    //[<Theory>]
    //[<InlineData("example1")>]
    //[<InlineData("input")>]
    //let Part2 (name:string) =
    //    let input, expected = GetPuzzleInput 2 name
    
    //    let actual = (new Puzzle()).RunPart2 input
    
    //    match expected with
    //    | None -> Assert.Null actual
    //    | _ -> Assert.Equal (expected.Value, actual)
    